# First Steps with Python and Jupyter 

![](https://i.imgur.com/gvSnw4A.png)



This tutorial covers the following topics:

* Performing arithmetic operations using Python
* Solving multi-step problems using variables
* Evaluating conditions using Python
* Combining conditions with logical operators
* Adding text styles using Markdown

#### Arithmetic operators and variables

In [1]:
#addition
2+3+5

10

In [2]:
#subtraction
7-8-3

-4

In [3]:
#Multiplication
4*3

12

In [4]:
#division
14/3

4.666666666666667

In [5]:
#To get only integer part of quotient(floor division)
14//3

4

In [6]:
# To get remainder
14%3

2

In [7]:
# exponent (5^3) 
5**3

125

* Solving multi-step problems using variables

Let's try solving the following word problem using Python: 

> A grocery store sells a bag of ice for $1.25 and makes a 20% profit. If it sells 500 bags of ice, how much total profit does it make?

We can list out the information provided and gradually convert the word problem into a mathematical expression that can be evaluated using Python. 

*Cost of ice bag ($)* = 1.25

*Profit margin* = 20% = .2

*Profit per bag ($)* = profit margin * cost of ice bag = .2 * 1.25

*No. of bags* = 500

*Total profit* = no. of bags * profit per bag = 500 * (.2 * 1.25)

In [8]:
500*(0.2*1.25)

125.0

Thus, the grocery store makes a total profit of $125. While this is a reasonable way to solve a problem, it's not entirely clear by looking at the code cell what the numbers represent. We can give names to each of the numbers by creating Python *variables*.

> **Variables**: While working with a programming language such as Python, information is stored in *variables*. You can think of variables as containers for storing data. The data stored within a variable is called its *value*.

In [9]:
cost_of_ice_bag=1.25
profit_margin = 0.2
number_of_bags = 500

In [10]:
profit_per_bag = cost_of_ice_bag*profit_margin

In [11]:
total_profit = number_of_bags*profit_per_bag

In [12]:
total_profit

125.0

Storing and manipulating data using appropriately named variables is a great way to explain what your code does.

Let's display the result of the word problem using a friendly message. We can do this using the `print` *function*.

> **Functions**: A function is a reusable set of instructions. It takes one or more inputs, performs certain operations, and often returns an output. Python provides many in-built functions like `print` and also allows us to define our own functions.

In [13]:
print("The grocery store makes a total profit of $", total_profit)

The grocery store makes a total profit of $ 125.0


Creating a code cell for each variable or mathematical operation can get tedious. Fortunately, Jupyter allows you to write multiple lines of code within a single code cell. The result of the last line of code within the cell is displayed as the output. 

Let's rewrite the solution to our word problem within a single cell.

In [14]:
# store input data in variables
cost_of_ice_bag=1.25
profit_margin = 0.2
number_of_bags = 500

#Perform the required calculations
profit_per_bag = cost_of_ice_bag*profit_margin
total_profit = number_of_bags*profit_per_bag

#display the result
print('The grocery store makes a total profit of $',total_profit)

The grocery store makes a total profit of $ 125.0


Note that we're using the `#` character to add *comments* within our code. 

> **Comments**: Comments and blank lines are ignored during execution, but they are useful for providing information to humans (including yourself) about what the code does. Comments can be inline (at the end of some code), on a separate line, or even span multiple lines. 

Inline and single-line comments start with `#`, whereas multi-line comments begin and end with three quotes, i.e. `'''` . Here are some examples of code comments:

In [15]:
my_favorite_number = 1 # an inline comment

# This comment gets its own line
my_least_favorite_number = 3

'''This is a multi-line comment.
Write as little or as much as you'd like.

Comments are really helpful for people reading
your code, but try to keep them short & to-the-point.

Also, if you use good variable names, then your code is
often self explanatory, and you may not even need comments!
'''
a_neutral_number = 5

#### Conditional operators

* Evaluating conditions using Python

Apart from arithmetic operations, Python also provides several operations for comparing numbers & variables.

| Operator    | Description                                                     |
|-------------|-----------------------------------------------------------------|
| `==`        | Check if operands are equal                                     |
| `!=`        | Check if operands are not equal                                 |
| `>`         | Check if left operand is greater than right operand             |
| `<`         | Check if left operand is less than right operand                |
| `>=`        | Check if left operand is greater than or equal to right operand |
| `<=`        | Check if left operand is less than or equal to right operand    |

The result of a comparison operation is either `True` or `False` (note the uppercase `T` and `F`). These are special keywords in Python. Let's try out some experiments with comparison operators.

In [16]:
my_favorite_number = 5
my_least_favorite_number = 3
a_neutral_number = 2

In [17]:
# Equality check
my_favorite_number == 1 , my_favorite_number==5

(False, True)

In [18]:
# Equality check
my_favorite_number ==  my_least_favorite_number

False

In [19]:
# Not equal check
my_favorite_number != a_neutral_number

True

In [20]:
# Not equal check
my_favorite_number!=5

False

In [21]:
# greater than check
my_favorite_number > 12

False

In [22]:
# less than check
my_favorite_number < 12

True

In [23]:
# greater than equal to check
my_favorite_number >= 12

False

In [24]:
# less than equal to check
my_favorite_number <= 12

True

In [25]:
#storing true or false in a variable

a=(my_favorite_number <= 12)
b=(my_favorite_number >= 12)
a,b

(True, False)

In [26]:
type(a),type(b)

(bool, bool)

#### Logical operators

* Combining conditions with logical operators

The logical operators `and`, `or` and `not` operate upon conditions and `True` & `False` values (also known as *booleans*). `and` and `or` operate on two conditions, whereas `not` operates on a single condition.

The `and` operator returns `True` when both the conditions evaluate to `True`. Otherwise, it returns `False`.

| `a`     | `b`    | `a and b` |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `False`   |
|  `False`| `True` | `False`   |
|  `False`| `False`| `False`   |

In [27]:
# and operator
(True and False), (False and True),(True and True),(False and False)

(False, False, True, False)

The `or` operator returns `True` if at least one of the conditions evaluates to `True`. It returns `False` only if both conditions are `False`.

| `a`     | `b`    | `a or b`  |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `True`    |
|  `False`| `True` | `True`    |
|  `False`| `False`| `False`   |


In [28]:
# or operator
(True or False), (False or True),(True or True),(False or False)

(True, True, True, False)

The `not` operator returns `False` if a condition is `True` and `True` if the condition is `False`.

In [29]:
# not operator
not True,not False

(False, True)

Logical operators can be combined to form complex conditions. Use round brackets or parentheses `(` and `)` to indicate the order in which logical operators should be applied.

In [30]:
(2>3 and 4<=5) or not(my_favorite_number<0 and True)

True

If parentheses are not used, logical operators are applied from left to right.

In [31]:
not True and 0<1 or False and True

False